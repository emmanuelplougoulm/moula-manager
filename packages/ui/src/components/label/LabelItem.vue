<template>
  <span
    id="label-item"
    class="text-small-semi-bold label--default"
    data-testid="label-item"
    :class="labelClass"
  >
    {{ text }}
    <TooltipItem
      v-if="tooltipTitle || tooltipDescription"
      :title="tooltipTitle"
      :description="tooltipDescription"
    />
  </span>
</template>

<script setup lang="ts">
import { computed } from 'vue';
import TooltipItem from '../tooltip/TooltipItem.vue';

const props = defineProps({
  text: {
    type: String,
    required: true
  },
  highlighted: {
    type: Boolean,
    required: true
  },
  tooltipTitle: {
    type: String,
    default: undefined
  },
  tooltipDescription: {
    type: String,
    default: undefined
  }
});

const labelClass = computed(() => ({
  [`label--highlighted`]: props.highlighted
}));
</script>

<style scoped>
.label--default {
  display: flex;
  justify-content: space-between;
  align-items: flex-end;
  color: var(--neutral-10);
  position: relative;
}
.label--highlighted {
  color: var(--primary-text);
}
</style>
