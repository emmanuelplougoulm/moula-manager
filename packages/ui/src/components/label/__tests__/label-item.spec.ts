import { mount } from '@vue/test-utils';
import LabelItem from '../LabelItem.vue';

describe('LabelItem tests', () => {
  it('should be highlighted', () => {
    const wrapper = mount(LabelItem, {
      props: {
        text: 'City of birth',
        highlighted: true
      }
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should not be highlighted', () => {
    const wrapper = mount(LabelItem, {
      props: {
        text: 'City of birth',
        highlighted: false
      }
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
