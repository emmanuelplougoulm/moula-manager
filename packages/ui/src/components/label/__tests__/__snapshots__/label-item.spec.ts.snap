// Vitest Snapshot v1

exports[`LabelItem tests > should be highlighted 1`] = `"<span data-v-0704e93f=\\"\\" id=\\"label-item\\" class=\\"text-small-semi-bold label--default label--highlighted\\" data-testid=\\"label-item\\">City of birth <!--v-if--></span>"`;

exports[`LabelItem tests > should not be highlighted 1`] = `"<span data-v-0704e93f=\\"\\" id=\\"label-item\\" class=\\"text-small-semi-bold label--default\\" data-testid=\\"label-item\\">City of birth <!--v-if--></span>"`;
